dali/test/python/test_operator_decoder.py

```diff
@@ -92,8 +94,24 @@ def check_FastDCT_body(batch_size, img_type, device):
                       # average difference should be no bigger by off-by-3
                       batch_size=batch_size, N_iterations=3, eps=3)
 
-def check_FastDCT():
+def test_FastDCT():
     for device in {'cpu', 'mixed'}:
         for batch_size in {1, 8}:
             for img_type in test_good_path:
               yield check_FastDCT_body, batch_size, img_type, device
+
+def test_image_decoder_memory_stats():
+    device = 'mixed'
+    img_type = 'jpeg'
+    def check(img_type, size, device, threads):
+        data_path = os.path.join(test_data_root, good_path, img_type)
+        # largest allocation should match our (in this case) memory padding settings
+        # (assuming no reallocation was needed here as the hint is big enough)
+        pattern = 'Device memory: \d+ allocations, largest = 16777216 bytes\n' + \
```

Comment:
It depends very much on the test data set. Maybe you should set some large enough value in the decoder and see if it matches?

Reply:
It comes from the default arguments:
```
  .AddOptionalArg("device_memory_padding",
      R"code(**`mixed` backend only** Padding for nvJPEG's device memory allocations in bytes.
This parameter helps to avoid reallocation in nvJPEG whenever a bigger image
is encountered and internal buffer needs to be reallocated to decode it.)code",
      16*1024*1024)
  .AddOptionalArg("host_memory_padding",
      R"code(**`mixed` backend only** Padding for nvJPEG's host memory allocations in bytes.
This parameter helps to avoid reallocation in nvJPEG whenever a bigger image
is encountered and internal buffer needs to be reallocated to decode it.)code",
      8*1024*1024)  // based on ImageNet heuristics (8MB)
```
Any reason to use different values in this test?

Reply:
Forgot about the default values. Then it is  okay.